django/db/migrations/utils.py

```diff
@@ -0,0 +1,28 @@
+from django.apps import apps
+from django.db.migrations.autodetector import MigrationAutodetector
+from django.db.migrations.loader import MigrationLoader
+from django.db.migrations.state import ProjectState
+
+
+def check_unmigrated_models(loader=None, connection=None):
+    """
+    :return: A (possibly empty) list of messages,
+    if the currently registered models have changes that are not
+    yet represented in a migration
+    """
+    messages = []
+    if loader is None:
+        loader = MigrationLoader(connection)
+
+    autodetector = MigrationAutodetector(
+        loader.project_state(),
+        ProjectState.from_apps(apps),
+    )
+    if autodetector.changes(graph=loader.graph):
+        messages.extend([
```

Comment:
As a follow up, in my testing two `.append()` calls are about 200ns faster than the one `.extend()` call, but I really should have spent that time looking at making the migrations Markus is already working on faster, as that is millions of times slower than the 200ns that could be saved here for Django's test suite alone.